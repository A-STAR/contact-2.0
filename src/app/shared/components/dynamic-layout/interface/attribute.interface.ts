import { IDynamicLayoutGenericItem, DynamicLayoutItemType } from './item.interface';

export interface IAttributeContext {
  debtId: number;
  personId: number;
  personRole: number;
  userId: number;
}

export enum AttributeType {
  FORMULA    = 'formula',
  FORMULA_ID = 'formulaId',
}

export interface IAttribute {
  type: AttributeType;
  value: string;
}

export interface IAttributeConfig {
  [key: string]: IAttribute[];
}

export interface IAttributePayload {
  attributes: IAttributeConfig;
  context: Partial<IAttributeContext>;
}

export interface IDynamicLayoutAttribute extends IDynamicLayoutGenericItem {
  attributeType: AttributeType;
  type: DynamicLayoutItemType.ATTRIBUTE;
  value: string;
  // Optional:
  dictCode?: number;
  lookupKey?: number;
}
