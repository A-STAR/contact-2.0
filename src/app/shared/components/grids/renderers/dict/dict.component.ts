import { Component } from '@angular/core';
import { ICellRendererParams } from 'ag-grid/main';
import { ICellRendererAngularComp } from 'ag-grid-angular';
import { Observable } from 'rxjs/Observable';
import { map } from 'rxjs/operators';

import { UserDictionariesService } from '@app/core/user/dictionaries/user-dictionaries.service';

@Component({
    selector: 'app-checkbox-cell-renderer',
    template: `{{ value$ | async }}`,
})
export class DictRendererComponent implements ICellRendererAngularComp {
  private params: ICellRendererParams;

  constructor(
    private userDictionariesService: UserDictionariesService,
  ) {}

  agInit(params: ICellRendererParams): void {
    this.params = params;
  }

  refresh(): boolean {
    return false;
  }

  get value$(): Observable<string> {
    const { value } = this.params;
    return this.userDictionariesService
      .getDictionary(this.params['dictCode'])
      .pipe(
        map(dict => dict[value] ? dict[value].name : value),
      );
  }
}
