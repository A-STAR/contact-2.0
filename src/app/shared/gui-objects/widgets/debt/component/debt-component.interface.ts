export interface IDebtComponent {
  id: number;
  typeCode: number;
  sum: number;
  currencyId: number;
}
