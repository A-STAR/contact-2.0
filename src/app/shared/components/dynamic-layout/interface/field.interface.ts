import { IDynamicLayoutGenericItem, DynamicLayoutItemType } from './item.interface';

export interface IDynamicLayoutField extends IDynamicLayoutGenericItem {
  type: DynamicLayoutItemType.FIELD;
}
