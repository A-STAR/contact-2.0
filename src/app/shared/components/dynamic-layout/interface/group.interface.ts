import { IDynamicLayoutGenericItem, DynamicLayoutItemType } from './item.interface';

export enum DynamicLayoutGroupType {
  PLAIN = 'plain',
  TABS  = 'tabs',
}

export interface IDynamicLayoutGroup extends IDynamicLayoutGenericItem {
  type: DynamicLayoutItemType.GROUP;
}
