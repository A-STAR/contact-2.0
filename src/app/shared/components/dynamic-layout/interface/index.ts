import { Observable } from 'rxjs/Observable';

import { IDynamicLayoutControl } from './control.interface';
import { IDynamicLayoutGroup } from './group.interface';
import { IDynamicLayoutTemplate } from './template.interface';

export * from './context.interface';
export * from './control.interface';
export * from './attribute.interface';
export * from './group.interface';
export * from './item.interface';
export * from './template.interface';

export type IDynamicLayoutItem = IDynamicLayoutGroup | IDynamicLayoutControl | IDynamicLayoutTemplate;

export interface IDynamicLayoutConfig {
  items: IDynamicLayoutItem[];
}

export interface IDynamicLayoutItemProperties {
  item: IDynamicLayoutItem;
  streams: {
    display: Observable<boolean>;
  };
}
