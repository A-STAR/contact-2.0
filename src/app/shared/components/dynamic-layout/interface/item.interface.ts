import { IContext } from './context.interface';

export enum DynamicLayoutItemType {
  CONTROL  = 'control',
  GROUP    = 'group',
  TEMPLATE = 'template',
}

export interface IDynamicLayoutGenericItem {
  display: IContext;
  type: DynamicLayoutItemType;
}
