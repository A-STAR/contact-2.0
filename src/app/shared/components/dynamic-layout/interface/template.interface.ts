import { IDynamicLayoutGenericItem, DynamicLayoutItemType } from './item.interface';

export interface IDynamicLayoutTemplate extends IDynamicLayoutGenericItem {
  type: DynamicLayoutItemType.TEMPLATE;
  value: string;
}
