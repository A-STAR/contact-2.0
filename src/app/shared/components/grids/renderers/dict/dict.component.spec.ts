import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Observable } from 'rxjs/Observable';
import { of } from 'rxjs/observable/of';

import { IUserTerm } from '@app/core/user/dictionaries/user-dictionaries.interface';

import { UserDictionariesService } from '@app/core/user/dictionaries/user-dictionaries.service';

import { DictRendererComponent } from './dict.component';

class UserDictionariesServiceMock {
  getDictionary(dictCode: number): Observable<IUserTerm[]> {
    const terms = Array(10).fill(null).map((_, i) => i + 1).map(code => ({
      code,
      name: `Term ${code}`,
      isClosed: false,
      parentCode: null,
      parentDictTerm: null,
    }));
    return of(terms);
  }
}

describe('DictRendererComponent', () => {
  let fixture: ComponentFixture<DictRendererComponent>;

  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        declarations: [
          DictRendererComponent,
        ],
        providers: [
          {
            provide: UserDictionariesService,
            useClass: UserDictionariesServiceMock,
          }
        ]
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DictRendererComponent);
  });

  it('should render dict renderer', () => {
    fixture.componentInstance.agInit({
      value: 1,
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
