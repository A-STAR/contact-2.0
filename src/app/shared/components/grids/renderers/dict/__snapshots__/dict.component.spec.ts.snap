// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictRendererComponent should render dict renderer 1`] = `
<div
  id="root0"
  ng-version="5.2.4"
>
  Term 2
</div>
`;
